### PR TITLE
TE Midpoint Review

### PR DESCRIPTION
**Pull request by fernandospr**
Summary of changes:
* Added missing Copyright to Kotlin files.
* Removed `.DS_Store` from all folders.
* Added `.gitignore` to all projects.
* Updated dependency versions.
* Added `TRANSPORT_CELLULAR` to `hasNetworkConnection` function (so that it also works with 3/4/5G networks).

### CODE REVIEW

File: 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/MainActivity.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ import androidx.compose.ui.Modifier
 import androidx.core.content.getSystemService
 import com.yourcompany.android.moviediary.networking.MovieDiaryApi
 import com.yourcompany.android.moviediary.ui.login.LoginScreen
-import com.yourcompany.android.moviediary.ui.movies.HomeScreen
```
Comment (LukaKordic) [inline 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/MainActivity.kt]: This change looks obsolete. I changed **home** package to **movies**, and `HomeScreen` to `MoviesScreen`.

File: 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/ui/home/HomeScreen.kt [hunk 2]
```diff
@@ -28,7 +28,7 @@
  * THE SOFTWARE.
  */
 
-package com.yourcompany.android.moviediary.ui.movies
```
Comment (LukaKordic) [inline 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/ui/home/HomeScreen.kt]: ⬆️

File: 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/MainActivity.kt (hunk 1 above)
Comment (fernandospr) [reply 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/MainActivity.kt]: This is from an old commit, you should see the final changes here: https://github.com/kodecocodes/m3-nra-materials/pull/2/files

File: 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/ui/home/HomeScreen.kt (hunk 2 above)
Comment (fernandospr) [reply 01-understand-network/Starter/app/src/main/java/com/yourcompany/android/moviediary/ui/home/HomeScreen.kt]: This is from an old commit, you should see the final changes here: https://github.com/kodecocodes/m3-nra-materials/pull/2/files

**Pull request merged**